ot/lp/emd_wrap.pyx

```diff
@@ -135,17 +77,13 @@ def emd2_c( np.ndarray[double, ndim=1, mode="c"] a,np.ndarray[double, ndim=1, mo
         b=np.ones((n2,))/n2
 
     # calling the function
-    cdef int resultSolver = EMD_wrap(n1,n2,<double*> a.data,<double*> b.data,<double*> M.data,<double*> G.data,<double*> &cost, max_iter)
+    cdef int resultSolver = EMD_wrap(n1,n2,<double*> a.data,<double*> b.data,<double*> M.data,<double*> G.data, <double*> alpha.data, <double*> beta.data, <double*> &cost, numItermax)
     if resultSolver != OPTIMAL:
         if resultSolver == INFEASIBLE:
-            print("Problem infeasible. Try to inscrease numItermax.")
+            warnings.warn("Problem infeasible. Check that a and b are in the simplex")
         elif resultSolver == UNBOUNDED:
-            print("Problem unbounded")
-
-    cost=0
-    for i in range(n1):
-        for j in range(n2):
-            cost+=G[i,j]*M[i,j]
-
-    return cost
+            warnings.warn("Problem unbounded")
+        elif resultSolver == MAX_ITER_REACHED:
+            warnings.warn("numItermax reached before optimality. Try to increase numItermax.")
 
+    return G, cost, alpha, beta
```

Comment:
could you also return the status integer so that on can detect from the log  a convergence problem instead of just printing the error?

It should be added also to the log I think